includes/class-wc-calypso-bridge-setup-checklist.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Setup Checklist
+ *
+ * Adds a new WC setup page with a checklist of steps for setting up your store.
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Calypso_Bridge_Admin_Setup_Checklist class.
+ */
+class WC_Calypso_Bridge_Admin_Setup_Checklist {
+
+	static $instance = false;
+
+	/**
+	 * Provide only a single instance of this class.
+	 */
+	public static function getInstance() {
+		if ( !self::$instance ) {
+			self::$instance = new self;
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Hooks into WordPress to add our new setup checklist.
+	 */
+	private function __construct() {
+
+		// If setup has been completed, do nothing.
+		if ( true === (bool) get_option( 'atomic-ecommerce-setup-checklist-complete', false ) ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'admin_menu' ) );
+
+		// priority is 20 to run after https://github.com/woocommerce/woocommerce/blob/a55ae325306fc2179149ba9b97e66f32f84fdd9c/includes/admin/class-wc-admin-menus.php#L165.
+		add_action( 'admin_head', array( $this, 'admin_menu_structure' ), 20 );
+
+		// run early and late
+		if ( isset( $_GET['page'] ) && 'wc-setup-checklist' === $_GET['page'] ) {
+			add_action( 'admin_notices', array( $this, 'hide_all_notices_on_setup_page' ), 0 );
```

Comment:
This is a cool solution.  Could we simplify this to `remove_all_actions( 'admin_notices' );` or is there something else this function does?

Reply:
Unfortunately, not all notices are hooked the same way (see https://github.com/Automattic/wc-calypso-bridge/pull/92/ for a few of the different ways that need done to selectively unhook things). I gave this a try to make sure, but I still had some notices come through -- so I think this might be the best way to hide everything.

Reply:
Can you try this?  Only tested with a couple but seemed to remove all my notices including those from Canada Post.

```php
	/**
	 * Hooks into WordPress to add our new setup checklist.
	 */
	private function __construct() {
		// If setup has been completed, do nothing.
		if ( true === (bool) get_option( 'atomic-ecommerce-setup-checklist-complete', false ) ) {
			return;
		}

		add_action( 'admin_menu', array( $this, 'admin_menu' ) );
		add_action( 'admin_head', array( $this, 'remove_notices' ) );
		// priority is 20 to run after https://github.com/woocommerce/woocommerce/blob/a55ae325306fc2179149ba9b97e66f32f84fdd9c/includes/admin/class-wc-admin-menus.php#L165.
		add_action( 'admin_head', array( $this, 'admin_menu_structure' ), 20 );
	}

	/**
	 * Remove all admin notices
	 */
	public function remove_notices() {
		remove_all_actions( 'admin_notices' );
	}
```